src/_posts/platform/app/2000-01-01-procfile.md

```diff
@@ -9,48 +9,159 @@ Procfile support is one of our
-each type of container for your app. It has to be located at the root of your project.
-Environment variables will be interpolated.
+### Special Process Types
 
-```yaml
-web: bundle exec rails server -p $PORT -e $RAILS_ENV -b 0.0.0.0
-worker: bundle exec sidekiq -e $RAILS_ENV
-```
+By default, Scalingo uses three process types: `web`, `tcp` and `postdeploy`.
+**These process types have specific use cases** and the platform will only use
+them for what they are designed for.
+
+#### The `web` Process Type
+
+The `web` process type defines how to start and run the Managed Runtime(s)
+handling HTTP requests.
+
+It's the only process type that can receive external HTTP requests from
+[Scalingo's routers]({% post_url platform/internals/2000-01-01-routing %}).
+Consequently, if your application has a web server, it MUST have a `web`
+process type bound to `$PORT` to be publicly reachable.
+
+The `web` process type is often the only one required.
+
+{% note %}
+- Almost all our [buildpacks]({% post_url platform/deployment/buildpacks/2000-01-01-intro %})
+  define a default `web` process type so you don't have to care about it.
+- The `web` process type is the only one that Scalingo scales to 1:M by
+  default. If you don't need it, (for example, if your application is a
+  [web-less application]({% post_url platform/app/2000-01-01-web-less-app %}))
+  you can scale it to 0 to avoid unwanted costs.
+{% endnote %}
 
-By default, when you first try to deploy your application, we'll try to
-launch the `web` process. We try to determine it without your help, but in
-some cases it is required to define it, as in the previous example. If your
-application requires a `Procfile` and you haven't defined it, your deployment
-will fail and a log entry will attest the lack of Procfile.
+#### The `tcp` Process Type
 
-The standard syntax is:
+The `tcp` process type defines how to start the
+[TCP Gateway addon]({% post_url addons/tcp-gateway/2000-01-01-start %}).
+
+#### The `postdeploy` Process Type
+
+The `postdeploy` process type defines how to start the
+[postdeploy hook]({% post_url platform/app/2000-01-01-postdeploy-hook %}).
+
+
+## Working With the Procfile
+
+### Procfile Naming Convention
+
+The Procfile is a plain text file that MUST be named `Procfile`. Even if it
+uses the YAML syntax, there is **no extension**: no `.txt`, no `.yaml`, no
+`.yml`. **Only `Procfile`**.
+
+The file MUST live in the application's root directory.
+
+### Defining a Process Type
+
+Each process type of the application is defined on its own, individual line,
+with the following format:
 
 ```yaml
-<container type>: <command>
+<process_type>: <command>
 ```
-where:
 
-- `<container type>` should contain letters and digits only.
-- `<command>` can be any Bash command. It can include environment variables.
+- `<process_type>` designates the name of the process type. It can only contain
+  alphanumerics, dashes and underscores (`[A-Za-z0-9_-]+`). Except this rule,
```

Comment:
Oh, I thought it was a mistake since it seems to be supported everywhere.
I removed it in https://github.com/Scalingo/documentation/pull/2020/commits/2996c740072b9fd503698e0a53e80840db89424d.
For my knowledge, is there a particular reason for this?

Reply:
Nope, initial implementation, it could be improved to get those characters.

For the `-` it's because our "log source" are using `-`, ie `[web-1] log line` so we don't allow `[worker-type-1-1]` since it may be complicated to read. For underscorre, maybe nothing would prevent us using it, I can't say right now